src/core/Akka/Actor/WatchAsyncSupport.cs

```diff
@@ -0,0 +1,58 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="WatchAsyncSupport.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Dispatch.SysMsg;
+using Akka.Util.Internal;
+
+namespace Akka.Actor
+{
+    /// <summary>
+    /// WatchAsync extensions
+    /// </summary>
+    public static class WatchAsyncSupport
+    {
+        /// <summary>
+        /// WatchAsync allows non-Akka.NET components to subscribe to DeathWatch notifications for a given <see cref="IActorRef"/>
+        /// </summary>
+        /// <param name="target">The actor to watch.</param>
+        /// <param name="token">Optional - a cancellation token.</param>
+        /// <returns><c>true</c> if the target was terminated, <c>false</c> otherwise.</returns>
+        /// <exception cref="InvalidOperationException">Thrown if we attempt to watch a</exception>
+        public static async Task<bool> WatchAsync(this IActorRef target, CancellationToken token = default)
```

Comment:
After I refactored the guts of `PromiseActorRef` implementing `WatchAsync` was pretty easy to implement - it's just the `GracefulStop` code but without sending a kill message to the actor _and_ it uses ` CancellationToken` instead of a `TimeSpan`. That's all.